ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -340,10 +340,11 @@ func (r *RayClusterReconciler) rayClusterReconcile(ctx context.Context, request
 	// Calculate the new status for the RayCluster. Note that the function will deep copy `instance` instead of mutating it.
```

Comment:
Can you validate the status of the RayCluster CR, `validateRayClusterStatus`, at the very beginning of the reconciliation to ensure that `RayClusterSuspending` and `RayClusterSuspended` are not both true at the same time? You can refer to [validateRayJobStatus](https://github.com/ray-project/kuberay/blob/bee1b71e01785aaa29c6ec9a0ad6604eb868579c/ray-operator/controllers/ray/rayjob_controller.go#L841) as an example.